gmso/core/topology.py

```diff
@@ -465,7 +482,47 @@ def update_connection_types(self):
                     c.connection_type = self._dihedral_types[c.connection_type]
                 if isinstance(c.connection_type, ImproperType):
                     c.connection_type = self._improper_types[c.connection_type]
+    def add_pairpotentialtype(self, pairpotentialtype):
```

Comment:
Just a comment, this method adds a `PairPotentialType` between two `AtomType`s only when they are in the topology. It might be nice to add a flag to `update_atom_types` just to provide and option to update the newly added Atom's  `atomtype` in the topology.

Reply:
Addressed most of the feedback from Umesh.


> 2. Should the added `PairPotentialType` be indexed? If so, the `get_index` method in `Topology` should be extended to incorporate `PairPotentialType` as well.

Regarding whether `PairPotentialType` should be indexed, I wonder if it is needed? we can already directly access a `PairPotentialType` by pairs of member `AtomType`s. As I am not fully sure about when to or not to index, please feel free to let me know your thought about this.